gitdash/pages/api/github.ts

```diff
@@ -1,58 +1,88 @@
+import parse from "parse-link-header";
+
+export default async function GetDetails(
+  req: any,
+  res: {
+    status: (arg0: number) => {
+      (): any;
+      new (): any;
+      json: {
+        (arg0: {
+          stars: any;
+          followers: any;
+          starred: any;
+          all_repos: any;
+          repo_count: any;
+          issues: any;
+          issue_count: any;
+        }): any;
+        new (): any;
+      };
+    };
+  }
+): Promise<any> {
+  const session = await getSession({ req });
 
-export default async (_req: any, res: { status: (arg0: number) => { (): any; new(): any; json: { (arg0: { stars: any; followers: any; starred: any; all_repos: any; repo_count: any; issues: any; issue_count: any }): any; new(): any } } }) => {
   const octokit = new Octokit({
-    // auth: 
-    // github token for a particular user, leaving empty for now
-  })
-  const username = "prakharrathi25"    // need to get the input from the user for this  
+    auth: session?.accessToken,
+  });
 
-  // Number of followers
-  const followers = await octokit.request(`/users/${username}/followers?per_page=100`)
-  const followerCount = followers.data.length
+  // Get user data
+  const userData = await octokit.request("GET /user");
+  const username = userData.data.login
 
-  // Get all repos 
-  const repos = await octokit.request(`/users/${username}/repos`);
+  // Get all repos
+  const repos = await octokit.request("GET /user/repos");
 
-  // Number of stars
-  const starsCount = repos.data.filter((repo: { fork: any }) => !repo.fork).reduce((acc: any, item: { stargazers_count: any }) => {
-      return acc + item.stargazers_count
-  }, 0)
+  // Number of followers
+  const followerCount = userData.data.followers;
 
-  // Number of starred repos
-  const reposStarred = await octokit.request(`/users/${username}/starred`)
-  const starredCount = reposStarred.data.length
+  // Number of stars
+  // Note this really needs some way of optimizing instead of getting all the repos, parsing
+  // and adding. GraphQL!!!!!!!! We need this optimisation in a lot of places 
+  const starsCount = repos.data
+    .filter((repo: { fork: any }) => !repo.fork)
+    .reduce((acc: any, item: { stargazers_count: any }) => {
+      return acc + item.stargazers_count;
+    }, 0);
 
-  // Number of issues 
-  const issueCount = repos.data.filter((repo: { fork: any }) => !repo.fork).reduce((acc: any, item: { open_issues: any }) => {
-    console.log(item.open_issues)
-    return acc + item.open_issues
-  }, 0)
+  // Making the request to get the response :/
+  const reposStarred = await octokit.request("GET /user/starred?per_page=1");
 
-  // List of repos 
-  const repo_names = repos.data.map((repo: { name: any }) => repo.name)
+  // Parsing the starred repos using a library
+  const linkHeader = reposStarred.headers.link;
+  const parsed = linkHeader ? parse(linkHeader) : parse('');
+  const starredCount = parsed?.last.page;
```

Comment:
Seeing you use the `?` here is almost enough to make me switch from JavaScript to TypeScript for all my personal projects